README.md

```diff
@@ -15,48 +15,162 @@ An out-of-tree GNU Radio block for communicating with the FlexRadio
+### Dependencies
+- Xcode Command Line Tools
+  ```
+  xcode-select --installed
+  ```
+- [Python](https://www.python.org/download/releases/2.7/)
+
+  It does require python 2.7 and although you probably definitely already have it installed on your system I went ahead and installed it again to make sure everything was up to date.
+
+- [XQuartz/X11](https://www.xquartz.org/)
+
+- GNU Radio
+
+  This is the part that kind of sucks. You no longer have MacPorts installed so the recommended install of GNU Radio by the group that made GNU Radio doesn't apply. There is a brew package that will install GNU Radio, `brew install gnuradio` but since brew dropped support for qt4 in favour of qt5 the install leaves you without GNU Radio Companion which we want. You can bugger around with installing qt4 or you can check out [this handy repo](https://github.com/cfriedt/gnuradio-for-mac-without-macports) which allows you to install it as a `.app`.
+
+- > pip (_Maybe not necessary. Try only setuptools below_)
+  > ```
+  > sudo easy_install pip
+  > ```
+
+- PKG-CONFIG
+  ```
+  brew install pkg-config
+  ```
+
+- glib 2
+  ```
+  brew install glib --universal
+  ```
+
+- Mono
+  ```
+  brew cask install mono-mdk
+  ```
+  ```
+  PKG_CONFIG_PATH=/Library/Frameworks/Mono.framework/Versions/{x.x.x}/lib/pkgconfig/
+  ```
+
+- setuptools
+  ```
+  brew install wget
+  wget https://bootstrap.pypa.io/get-pip.py
+  sudo -H python get-pip.py
+  ```
+
+- PythonNet **(if the following doesn't work, see the [troubleshooting](#troubleshooting) section)**
+  ```
+  git clone https://github.com/pythonnet/pythonnet
+  cd pythonnet/
+  ```
+  ```
+  python setup.py bdist_wheel
+  ```
+  > Look inside the newly created dist/ folder for the details of this command
+  ```
+  sudo pip install --user dist/pythonnet-{pythonnet version}-{cpython version}-{mac veresion}.whl
+  ```
+
+- cmake
+  ```
+  brew install cmake
+  ```
+
+- boost
+  ```
+  brew install boost
+  ```
+
+- doxygen
+  ```
+  brew install doxygen
+  ```
 
-- [PythonNet] **(if the following doesn't work, see the [troubleshooting](#troubleshooting) section)**
+### Installing the Block
+
+Once all of the pre-requisites are installed, you can run the build script within the root of the project. A slight tweak needs to be made if GNU Radio was installed using PyBombs. For a non-pybombs install:
 ```
-sudo -H pip install pythonnet
+cd REPO_DIRECTORY
+sudo ./build.sh
 ```
 
-- [cmake](cmake.org)
+For a PyBombs install of GNU Radio, you need to open the script and comment out (prepend with a #) the following few lines near the bottom which say:
 ```
-sudo apt-get install cmake
+cmake ../
+make
+sudo make install
 ```
 
-### Installing the Block
-
-Once all of the pre-requisites are installed, you can run the build script within the root of the project.
-
+Then to install:
 ```
-cd <repo_director>
+cd REPO_DIRECTORY
 sudo ./build.sh
+cd build/
+cmake -DCMAKE_INSTALL_PREFIX=/FULL/PATH/TO/PYBOMBS/PREFIX ../
```

Comment:
We could definitely add a parameter to the build.sh to account for this so that we don't have to build twice